assets/eip-5058/test/test.ts

```diff
@@ -0,0 +1,157 @@
+  });
+  
+  it("isLocked works", async function() {
+    const NFTId = 0;
+    const block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await EIP5058.lockMint(owner.address, NFTId, timestamp + 2);
+    
+    // isLocked works
+    expect(await EIP5058.isLocked(NFTId)).eq(true);
+    await ethers.provider.send("evm_mine", []);
+    expect(await EIP5058.isLocked(NFTId)).eq(false);
+  });
+  
+  it("lock works", async function() {
+    const NFTId = 0;
+    let block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await EIP5058.lockMint(owner.address, NFTId, timestamp + 3);
+    
+    await expect(EIP5058.lock(NFTId, timestamp + 5)).to.be.revertedWith(
+      "EIP5058: token is locked",
+    );
+    
+    await ethers.provider.send("evm_mine", []);
+    await EIP5058.lock(NFTId, timestamp + 5);
+  });
+  
+  it("unlock works with lockMint", async function() {
+    const NFTId = 0;
+    const block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await EIP5058.lockMint(owner.address, NFTId, timestamp + 3);
+    
+    // unlock works
+    expect(await EIP5058.isLocked(NFTId)).eq(true);
+    expect(await EIP5058.lockerOf(NFTId)).eq(owner.address);
+    await EIP5058.unlock(NFTId);
+    expect(await EIP5058.isLocked(NFTId)).eq(false);
+  });
+  
+  it("unlock works", async function() {
+    const NFTId = 0;
+    
+    await EIP5058.mint(owner.address, NFTId);
+    
+    await expect(EIP5058.unlock(NFTId)).to.be.revertedWith(
+      "EIP5058: locker query for non-locked token",
+    );
+    const block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await EIP5058.lock(NFTId, timestamp + 3);
+    expect(await EIP5058.isLocked(NFTId)).eq(true);
+    await EIP5058.unlock(NFTId);
+    expect(await EIP5058.isLocked(NFTId)).eq(false);
+  });
+  
+  it("lockApprove works", async function() {
+    const NFTId = 0;
+    await EIP5058.mint(alice.address, NFTId);
+    
+    let block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await expect(EIP5058.lock(NFTId, timestamp + 2)).to.be.revertedWith(
+      "EIP5058: lock caller is not owner nor approved",
+    );
+    
+    await EIP5058.connect(alice).lockApprove(owner.address, NFTId);
+    expect(await EIP5058.getLockApproved(NFTId)).eq(owner.address);
+    
+    await expect(EIP5058.lock(NFTId, timestamp + 4)).to.be.revertedWith(
+      "EIP5058: lock from incorrect owner",
+    );
+    await EIP5058.lock(NFTId, timestamp + 6);
```

Comment:
The transaction sender seems the same, but the first lock reverts and the second lock succeeds?